server/src/service.rs

```diff
@@ -125,15 +128,46 @@ impl Service {
             .name(oo7::dbus::api::Service::DESTINATION.as_deref().unwrap())?
             .build()
             .await?;
+        let object_server = connection.object_server();
         let service = Self {
+            collections: Mutex::new(Vec::new()),
             manager: Default::default(),
             connection: connection.clone(),
         };
+        let collections = service.init_collections().await;
 
-        connection
-            .object_server()
+        object_server
             .at(oo7::dbus::api::Service::PATH.as_deref().unwrap(), service)
             .await?;
+
+        for collection in collections {
+            object_server
+                .at(collection.path().clone(), collection)
+                .await?;
+        }
+
         Ok(())
     }
+
+    async fn init_collections(&self) -> Vec<Collection> {
+        let mut collections: Vec<Collection> = Vec::new();
+        // todo: create default collection
+
+        // create temporary session collection
+        let session = Collection::new(
+            "session".to_string(),
+            "session".to_string(),
+            Arc::clone(&self.manager),
+        );
+        collections.push(session);
+
+        for collection in &collections {
+            self.collections
```

Comment:
Lock once before for instead of doing that for every iteration...

Reply:
understood.